packages/expo-cli/src/prompts.ts

```diff
@@ -0,0 +1,34 @@
+import program from 'commander';
+import prompts, { Choice, PromptObject, PromptType } from 'prompts';
+
+import CommandError from './CommandError';
+
+export { PromptType, PromptObject };
+
+type CliPrompts = PromptObject | PromptObject[];
+
+export default function prompt(
+  cliPrompts: CliPrompts,
+  { nonInteractiveHelp }: { nonInteractiveHelp?: string } = {}
+) {
+  const nPrompts = Array.isArray(cliPrompts) ? cliPrompts.length : 1;
+  if (program.nonInteractive && nPrompts !== 0) {
+    let message = `Input is required, but Expo CLI is in non-interactive mode.\n`;
+    if (nonInteractiveHelp) {
+      message += nonInteractiveHelp;
+    } else {
+      const question: any = Array.isArray(cliPrompts) ? cliPrompts[0] : cliPrompts;
+      message += `Required input:\n${(question.message || '').trim().replace(/^/gm, '> ')}`;
+    }
+    throw new CommandError('NON_INTERACTIVE', message);
+  }
+  return prompts(cliPrompts, {
+    onCancel() {
+      throw new CommandError('ABORTED');
```

Comment:
I had to implement this by default. `prompts` allows the user to "cancel" input and continue the execution. This is not how all code was built, adding this would make refactoring _a lot easier_. After the initial refactoring, we can do another iteration to remove this ofc!

Reply:
I think this makes sense. If some code requires cancellation, it can either catch this error (can add a `code` property it then to make it distinguishable from other errors) or we can allow passing `onCancel` to this function.

Either way, I think this is fine for now!

Reply:
Yeah, I think adding the prompt's `Option` interface would be great. With that, you can overwrite the default `onCancel` behavior by simply using `prompts([], { onCancel() { } });` If this is too verbose, we could also add a `abortOnCancel: false` option?

Reply:
`onCancel` seems fine and is more flexible